### PR TITLE
esp-hal: rmt: make ChannelCreator public

### PR DESCRIPTION
**Pull request by liebman**
## Thank you for your contribution!

We appreciate the time and effort you've put into this pull request.
To help us review it efficiently, please ensure you've gone through the following checklist:

### Submission Checklist 📝
- [x] I have updated existing examples or added new ones (if applicable).
- [x] I have used `cargo xtask fmt-packages` command to ensure that all changed code is formatted correctly.
- [x] My changes were added to the [`CHANGELOG.md`](https://github.com/esp-rs/esp-hal/blob/main/esp-hal/CHANGELOG.md) in the **_proper_** section.
- [x] My changes are in accordance to the [esp-rs API guidelines](https://github.com/esp-rs/esp-hal/blob/main/API-GUIDELINES.md)

#### Extra:
- [x] I have read the [CONTRIBUTING.md guide](https://github.com/esp-rs/esp-hal/blob/main/CONTRIBUTING.md) and followed its instructions.

#### Description
When using an Rmt channel you access `rmt.channel0` to grab channel 0.  However its type `ChannelCreator<>` is not public and can't be named as a function arg or struct member.  This PR makes that public.

#### Testing
Compiled/Run in a local project where I add the rmt channels as struct members.

### CODE REVIEW

File: esp-hal/src/rmt.rs [hunk 1]
```diff
@@ -684,6 +684,8 @@ mod impl_for_chip {
         }
     }
 
+    /// RMT Channel Creator
+    #[allow(missing_docs)]
```
Comment (bjoernQ) [inline esp-hal/src/rmt.rs:688]: We probably should have _some_ documentation here

Comment (liebman) [reply esp-hal/src/rmt.rs:688]: Yea - I followed exactly what `struct Rmt` has just above. Thoughts on what it should include?

Comment (bjoernQ) [reply esp-hal/src/rmt.rs:688]: Ah sorry .... it was too early when I looked at this, I guess.

**Review by bjoernQ — APPROVED** [review]
LGTM

**Comment by jessebraham** [comment]
(Sorry I didn't mean to click that button, can't believe there's no confirmation, hopefully I didn't break anything 🤣)

**Review by JurajSadel — APPROVED** [review]
LGTM, thanks!

**Review by jessebraham — APPROVED** [review]
Thanks!

**Pull request merged**